Camera-Controls-Java/app/src/main/java/com/tokbox/sample/cameracontrols/MainActivity.java

```diff
@@ -0,0 +1,321 @@
+        }
+    }
+
+    /* Make a request for session data */
+    private void getSession() {
+        Log.i(TAG, "getSession");
+
+        Call<GetSessionResponse> call = apiService.getSession();
+
+        call.enqueue(new Callback<GetSessionResponse>() {
+            @Override
+            public void onResponse(Call<GetSessionResponse> call, Response<GetSessionResponse> response) {
+                GetSessionResponse body = response.body();
+                initializeSession(body.apiKey, body.sessionId, body.token);
+            }
+
+            @Override
+            public void onFailure(Call<GetSessionResponse> call, Throwable t) {
+                throw new RuntimeException(t.getMessage());
+            }
+        });
+    }
+
+    private void initializeSession(String apiKey, String sessionId, String token) {
+        Log.i(TAG, "apiKey: " + apiKey);
+        Log.i(TAG, "sessionId: " + sessionId);
+        Log.i(TAG, "token: " + token);
+
+        /*
+        The context used depends on the specific use case, but usually, it is desired for the session to
+        live outside of the Activity e.g: live between activities. For a production applications,
+        it's convenient to use Application context instead of Activity context.
+         */
+        session = new Session.Builder(this, apiKey, sessionId).build();
+        session.setSessionListener(sessionListener);
+        session.connect(token);
+    }
+
+    private void initRetrofit() {
+        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
+        logging.setLevel(Level.BODY);
+
+        OkHttpClient client = new OkHttpClient.Builder()
+                .addInterceptor(logging)
+                .build();
+
+        retrofit = new Retrofit.Builder()
+                .baseUrl(ServerConfig.CHAT_SERVER_URL)
+                .addConverterFactory(MoshiConverterFactory.create())
+                .client(client)
+                .build();
+
+        apiService = retrofit.create(APIService.class);
+    }
+
+    private void finishWithMessage(String message) {
+        Log.e(TAG, message);
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
+        this.finish();
+    }
+
+    boolean cameraTorch = false;
+    public void SetTorch(View view) {
+        cameraTorch = !cameraTorch;
+        publisher.setCameraTorch(cameraTorch);
+        if(cameraTorch) {
+            buttonTorch.setBackgroundTintList(ColorStateList.valueOf(Color.RED));
+        } else {
+            buttonTorch.setBackgroundTintList(ColorStateList.valueOf(Color.GREEN));
+        }
+    }
+
+    float zoomFactor = 1.0f;
+    public void SetZoom(View view) {
+        if(zoomFactor == 0.5f) {
+            zoomFactor = 1.0f;
+        } else if(zoomFactor == 1.0f) {
+            zoomFactor = 5.0f;
+        } else {
+            zoomFactor = 0.5f;
+        }
+
+        publisher.setCameraZoomFactor(zoomFactor);
```

Comment:
This a way to set some zoom value. Implementation for production app might be different.

Reply:
Exactly. This is an example which will use both ultra-wide-zoom and the zoom-in. Do you want me to chnge something?